matadi/_templates.py

```diff
@@ -109,15 +117,21 @@ def __init__(self, fun, **kwargs):
         self.fun = fun
         self.kwargs = kwargs
         self.W = Material(self.x, self._fun_wrapper, kwargs=self.kwargs)
-        self.function = self.W.function
-        self.gradient = self.W.gradient
-        self.hessian = self.W.hessian
         self.gradient_vector_product = self.W.gradient_vector_product
         self.hessian_vector_product = self.W.hessian_vector_product
 
     def _fun_wrapper(self, x, **kwargs):
         return self.fun(x[0], **kwargs)
 
+    def function(self, x, *args, **kwargs):
+        return self.W.function(x[:1], *args, **kwargs)
+
+    def gradient(self, x, *args, **kwargs):
+        return [*self.W.gradient(x[:1], *args, **kwargs), None]
+
+    def hessian(self, x, *args, **kwargs):
+        return self.W.hessian(x[:1], *args, **kwargs)
+
```

Comment:
This is the essential change: Take only the first item of `x` as state variables are unused within the hyperelastic materials.

**Hint**: No change is made to the attribute `.x` (does not include state variables).